File: benchmarks/cpp/integer_boolean/big_integer/Divide.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BBigInteger.cpp"
#include "BBoolean.cpp"

#ifndef Divide_H
#define Divide_H

using namespace std;

class Divide {



    private:



        BBigInteger counter;
        BBigInteger value;

    public:

        Divide() {
            counter = static_cast<BBigInteger >((BBigInteger("0")));
            value = static_cast<BBigInteger >((BBigInteger("0")));
        }

        void simulate() {
            while((counter.less((BBigInteger("5000000")))).booleanValue()) {
                counter = static_cast<BBigInteger >(counter.plus((BBigInteger("1"))));
                value = static_cast<BBigInteger >(value.divide((BBigInteger("1"))));
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    Divide exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

File: benchmarks/java/integer_boolean/Clojure_Big_Integer/Divide.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class Divide {






    private BInteger counter;
    private BInteger value;

    public Divide() {
        counter = new BInteger("0");
        value = new BInteger("0");
    }

    public void simulate() {
        while((counter.less(new BInteger("5000000"))).booleanValue()) {
            counter = counter.plus(new BInteger("1"));
            value = value.divide(new BInteger("1"));
        }
    }

    public static void main(String[] args) {
        Divide exec = new Divide();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));

        Runtime runtime = Runtime.getRuntime();
        long memory = runtime.totalMemory() - runtime.freeMemory();
        System.out.println("Used memory is bytes: " + memory);
        System.out.println("Used memory is megabytes: "
                + (memory / (1024L * 1024L)));
    }

}
